### ISSUE TITLE
Unrecoverable error: WorkerLostError('Could not start worker processes',)

### ISSUE THREAD

**Issue by MattBlack85** [celery/celery#2966]
I recently updated the codebase of one project to django 1.8, the project is making heavy use of celery so I bumped also celery to the latest version but I have a strange problem.
If I run `celery worker -l info` the worker starts and of course it does nothing, but when I try to load the configuration from my celery.py file (is the same from the docs) I get errors while workers start:
celery --app=myapp worker -l info
```
2015-12-18 13:29:43,969 celery.worker.start/ERROR] Unrecoverable error: WorkerLostError('Could not start worker processes',)
Traceback (most recent call last):
  File "/home/matt/repos/Main/venv/local/lib/python2.7/site-packages/celery/worker/__init__.py", line 206, in start
    self.blueprint.start(self)
  File "/home/matt/repos/Main/venv/local/lib/python2.7/site-packages/celery/bootsteps.py", line 123, in start
    step.start(parent)
  File "/home/matt/repos/Main/venv/local/lib/python2.7/site-packages/celery/bootsteps.py", line 374, in start
    return self.obj.start()
  File "/home/matt/repos/Main/venv/local/lib/python2.7/site-packages/celery/worker/consumer.py", line 278, in start
    blueprint.start(self)
  File "/home/matt/repos/Main/venv/local/lib/python2.7/site-packages/celery/bootsteps.py", line 123, in start
    step.start(parent)
  File "/home/matt/repos/Main/venv/local/lib/python2.7/site-packages/celery/worker/consumer.py", line 821, in start
    c.loop(*c.loop_args())
  File "/home/matt/repos/Main/venv/local/lib/python2.7/site-packages/celery/worker/loops.py", line 48, in asynloop
    raise WorkerLostError('Could not start worker processes')
WorkerLostError: Could not start worker processes
```
Another strage thing is that I tried to comment out `app.config_from_object` and `app.autodiscover_tasks` thinking it could be a settings problem but I still get the WorkerLostError.

version used:
celery 3.1.19
billiard 3.3.0.22
amqp 1.4.6
kombu 3.0.32
django 1.8.7

The only thing I know is that `obj.pool.did_start_ok()` is returning `False`, but I have no clue why

**Comment by MattBlack85** [comment]
seems that giving a `-P gevent` make this work...

**Comment by ask** [comment]
Must be something with your configuration, but I don't have any guess.  It works fine for me here.

Try running the example in the documentation, then add things from your app until it breaks ;)

**Comment by astawiarski** [comment]
@MattBlack85 got the same yesterday when running inside docker container. I was using regular multiprocessing there (20 workers) and exceeding assigned memory (quite large app). Increasing the memory solved the problem (although my final solution was to reduce workers and have multiple containers instead)

**Comment by ask** [comment]
Closing this, as we don't have the resources to complete this task.

**Comment by bradmwalker** [comment]
This traceback occurred for me with an unchecked exception in a worker_process_init.connect handler using Celery 3.1.23.

**Comment by sunilx64** [comment]
I have faced the same issue in a docker container, it was happening because of resources(memory) was provided to container. After increasing the memory it started working as expected.

**Comment by MCDELTAT** [comment]
I have some additional clarifications for those stumbling on this like I did. TL;DR think of it as memory per worker thread.

I was spinning up a project that uses Celery via a helm chart. They provided an option to specify memory first, so I bumped that up. The pods ran a little better, but would still fail often. I realized they had a value for concurrent workers set to 0, which meant it would use all cores on the system. 

The pod was on a physical server with 96 cores, so it created 96 workers. I had specified 8GB of memory, so each worker was only getting ~85M of RAM. I dropped it to 20 workers, or about 400M per worker. This resulted in a much more stable system.
